### ISSUE TITLE
Fail on 2025 compiler

### ISSUE THREAD

**Issue by ZzEeKkAa** [intel/torch-xpu-ops#1027]
### 🐛 Describe the bug

While building pytorch using DLE with 2025.0.0 compiler:

```
cd /home/jovyan/Projects/pytorch/build/caffe2/aten_xpu/src/CMakeFiles/torch_xpu_ops_sycl_kernels.dir/ATen/native/xpu/sycl && /home/jovyan/Projects/triton/.venv/lib/python3.9/site-packages/cmake/data/bin/cmake -E make_directory /home/jovyan/Projects/pytorch/build/caffe2/aten_xpu/src/CMakeFiles/torch_xpu_ops_sycl_kernels.dir/ATen/native/xpu/sycl/. && /home/jovyan/Projects/triton/.venv/lib/python3.9/site-packages/cmake/data/bin/cmake -D verbose:BOOL=OFF -D generated_file:STRING=/home/jovyan/Projects/pytorch/build/caffe2/aten_xpu/src/CMakeFiles/torch_xpu_ops_sycl_kernels.dir/ATen/native/xpu/sycl/./torch_xpu_ops_sycl_kernels_generated_TensorModeKernel.cpp.o -P /home/jovyan/Projects/pytorch/build/caffe2/aten_xpu/src/CMakeFiles/torch_xpu_ops_sycl_kernels.dir/ATen/native/xpu/sycl/torch_xpu_ops_sycl_kernels_generated_TensorModeKernel.cpp.o.Release.cmake
FAILED: caffe2/aten_xpu/src/CMakeFiles/torch_xpu_ops_sycl_kernels.dir/ATen/native/xpu/sycl/torch_xpu_ops_sycl_kernels_generated_TensorModeKernel.cpp.o /home/jovyan/Projects/pytorch/build/caffe2/aten_xpu/src/CMakeFiles/torch_xpu_ops_sycl_kernels.dir/ATen/native/xpu/sycl/torch_xpu_ops_sycl_kernels_generated_TensorModeKernel.cpp.o
cd /home/jovyan/Projects/pytorch/build/caffe2/aten_xpu/src/CMakeFiles/torch_xpu_ops_sycl_kernels.dir/ATen/native/xpu/sycl && /home/jovyan/Projects/triton/.venv/lib/python3.9/site-packages/cmake/data/bin/cmake -E make_directory /home/jovyan/Projects/pytorch/build/caffe2/aten_xpu/src/CMakeFiles/torch_xpu_ops_sycl_kernels.dir/ATen/native/xpu/sycl/. && /home/jovyan/Projects/triton/.venv/lib/python3.9/site-packages/cmake/data/bin/cmake -D verbose:BOOL=OFF -D generated_file:STRING=/home/jovyan/Projects/pytorch/build/caffe2/aten_xpu/src/CMakeFiles/torch_xpu_ops_sycl_kernels.dir/ATen/native/xpu/sycl/./torch_xpu_ops_sycl_kernels_generated_TensorModeKernel.cpp.o -P /home/jovyan/Projects/pytorch/build/caffe2/aten_xpu/src/CMakeFiles/torch_xpu_ops_sycl_kernels.dir/ATen/native/xpu/sycl/torch_xpu_ops_sycl_kernels_generated_TensorModeKernel.cpp.o.Release.cmake
/home/jovyan/Projects/pytorch/third_party/torch-xpu-ops/src/ATen/native/xpu/sycl/TensorModeKernel.cpp:796:46: error: expected '(' for function-style cast or type construction
  796 |         default_sorter<std::greater<scalar_t>>::template memory_required<
      |                        ~~~~~~~~~~~~~~~~~~~~~~^
/home/jovyan/Projects/pytorch/third_party/torch-xpu-ops/src/ATen/native/xpu/sycl/TensorModeKernel.cpp:796:9: error: no member named 'default_sorter' in namespace 'sycl::ext::oneapi::experimental'
  795 |     auto sort_scratch_memory_size = sycl::ext::oneapi::experimental::
      |                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  796 |         default_sorter<std::greater<scalar_t>>::template memory_required<
      |         ^
/home/jovyan/Projects/pytorch/third_party/torch-xpu-ops/src/ATen/native/xpu/sycl/TensorModeKernel.cpp:796:58: error: no member named 'memory_required' in the global namespace
  796 |         default_sorter<std::greater<scalar_t>>::template memory_required<
      |                                               ~~         ^
3 errors generated.
CMake Error at torch_xpu_ops_sycl_kernels_generated_TensorModeKernel.cpp.o.Release.cmake:145 (message):
  Error generating file
  /home/jovyan/Projects/pytorch/build/caffe2/aten_xpu/src/CMakeFiles/torch_xpu_ops_sycl_kernels.dir/ATen/native/xpu/sycl/./torch_xpu_ops_sycl_kernels_generated_TensorModeKernel.cpp.o
```

### Versions

```
Collecting environment information...
PyTorch version: N/A
Is debug build: N/A
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.5 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: Could not collect
CMake version: version 3.30.3
Libc version: glibc-2.35

Python version: 3.9.20 (main, Sep  7 2024, 18:35:25)  [GCC 11.4.0] (64-bit runtime)
Python platform: Linux-5.15.0-118-generic-x86_64-with-glibc2.35
Is CUDA available: N/A
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: N/A

CPU:
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        52 bits physical, 57 bits virtual
Byte Order:                           Little Endian
CPU(s):                               128
On-line CPU(s) list:                  0-127
Vendor ID:                            GenuineIntel
Model name:                           Intel(R) Xeon(R) Gold 6438Y+
CPU family:                           6
Model:                                143
Thread(s) per core:                   2
Core(s) per socket:                   32
Socket(s):                            2
Stepping:                             8
CPU max MHz:                          4000.0000
CPU min MHz:                          800.0000
BogoMIPS:                             4000.00
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cat_l2 cdp_l3 invpcid_single cdp_l2 ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect avx_vnni avx512_bf16 wbnoinvd dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req avx512vbmi umip pku ospke waitpkg avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid bus_lock_detect cldemote movdiri movdir64b enqcmd fsrm md_clear serialize tsxldtrk pconfig arch_lbr amx_bf16 avx512_fp16 amx_tile amx_int8 flush_l1d arch_capabilities
Virtualization:                       VT-x
L1d cache:                            3 MiB (64 instances)
L1i cache:                            2 MiB (64 instances)
L2 cache:                             128 MiB (64 instances)
L3 cache:                             120 MiB (2 instances)
NUMA node(s):                         2
NUMA node0 CPU(s):                    0-31,64-95
NUMA node1 CPU(s):                    32-63,96-127
Vulnerability Gather data sampling:   Not affected
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Not affected
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Not affected
Vulnerability Spec rstack overflow:   Not affected
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI BHI_DIS_S
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected

Versions of relevant libraries:
[pip3] numpy==2.0.2
[pip3] optree==0.12.1
[pip3] triton==3.1.0
[pip3] triton-kernels-benchmark==3.1.0+git52f279a3
[conda] No relevant packages
```

**Comment by ZzEeKkAa** [comment]
Fail is introduced here https://github.com/intel/torch-xpu-ops/pull/770 because of https://github.com/intel/llvm/pull/11863

**Comment by fengyuan14** [comment]
Thanks for your report and fixing. We have got a PR about it. https://github.com/intel/torch-xpu-ops/pull/1017

**Comment by ZzEeKkAa** [comment]
Thank you! Fixed in #1017
